### PR TITLE
More updates for Data Prepper 2.0

### PR DESCRIPTION
**Pull request by oeyh**
### Description

Additional updates for Data Prepper 2.0 documentation - adds missing options and fixes.

Targets at `data-prepper-2.0` branch for the coming release.

### Issues Resolved
N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _clients/data-prepper/data-prepper-reference.md [hunk 1]
```diff
@@ -19,8 +19,11 @@ keyStorePassword | No | String | Password for keystore. Optional, defaults to em
 privateKeyPassword | No | String | Password for private key within keystore. Optional, defaults to empty string.
 serverPort | No | Integer | Port number to use for server APIs. Defaults to 4900
 metricRegistries | No | List | Metrics registries for publishing the generated metrics. Currently supports Prometheus and CloudWatch. Defaults to Prometheus.
+metricTags | No | Map | Key-value pairs as common metric tags to metric registries. The maximum number of pairs is 3. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Its value could also be set through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, the value will overwrite those set through the above methods.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:22]: > Its value could also be set through the

I recommend using active voice.

> Alternatively, administrators can set this value through the

Comment (Naarcha-AWS) [inline _clients/data-prepper/data-prepper-reference.md:22]:
```suggestion
metricTags | No | Map | Key-value pairs as common metric tags to metric registries. The maximum number of pairs is three. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Its value could also be set through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, that value overwrites those set through the above methods.
```

File: _clients/data-prepper/data-prepper-reference.md [hunk 2]
```diff
@@ -90,6 +94,7 @@ Option | Required | Type | Description
 port | No | Integer | The port OTel trace source is running on. Default is `21890`.
 request_timeout | No | Integer | The request timeout in milliseconds. Default is `10_000`.
 health_check_service | No | Boolean | Enables a gRPC health check service under `grpc.health.v1/Health/Check`. Default is `false`.
+unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. This option is ignored if no authentication is defined. Default is `false`.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:97]: > Whether authentication is required

Perhaps add a verb here:

> Determines whether authentication is required

File: _clients/data-prepper/data-prepper-reference.md [hunk 3]
```diff
@@ -19,8 +19,11 @@ keyStorePassword | No | String | Password for keystore. Optional, defaults to em
 privateKeyPassword | No | String | Password for private key within keystore. Optional, defaults to empty string.
 serverPort | No | Integer | Port number to use for server APIs. Defaults to 4900
 metricRegistries | No | List | Metrics registries for publishing the generated metrics. Currently supports Prometheus and CloudWatch. Defaults to Prometheus.
+metricTags | No | Map | Key-value pairs as common metric tags to metric registries. The maximum number of pairs is 3. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Its value could also be set through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, the value will overwrite those set through the above methods.
+authentication | No | Object | Authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server will not perform authentication.
```
Comment (Naarcha-AWS) [inline _clients/data-prepper/data-prepper-reference.md:23]:
```suggestion
authentication | No | Object | Authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server does not perform authentication.
```

File: _clients/data-prepper/data-prepper-reference.md [hunk 4]
```diff
@@ -109,6 +114,8 @@ This is a source plugin that supports HTTP protocol. Currently ONLY support Json
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 port | No | Integer | The port the source is running on. Default is `2021`. Valid options are between `0` and `65535`.
+health_check_service | No | Boolean | Enables health check service on `/health` endpoint on the defined port. Default is `false`.
+unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. This option is ignored if no authentication is defined. Default is `false`.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:118]: Similar comment as above.

File: _clients/data-prepper/data-prepper-reference.md (hunk 2 above)
Comment (Naarcha-AWS) [inline _clients/data-prepper/data-prepper-reference.md:97]:
```suggestion
unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.
```

File: _clients/data-prepper/data-prepper-reference.md [hunk 5]
```diff
@@ -472,12 +480,10 @@ socket_timeout | No | Integer | the timeout in milliseconds for waiting for data
...
-trace_analytics_raw | No | Boolean | Deprecated in favor of `index_type`. Whether to export as trace data to the `otel-v1-apm-span-*` index pattern (alias `otel-v1-apm-span`) for use with the Trace Analytics OpenSearch Dashboards plugin. Default is `false`.
-trace_analytics_service_map | No | Boolean | Deprecated in favor of `index_type`. Whether to export as trace data to the `otel-v1-apm-service-map` index for use with the service map component of the Trace Analytics OpenSearch Dashboards plugin. Default is `false`.
-index | No | String | Name of the index to export to. Only required if you don't use the `trace-analytics-raw` or `trace-analytics-service-map` presets. In other words, this parameter is applicable and required only if index_type is explicitly `custom` or defaults to `custom`.
+index | Conditionally | String | Name of the index to export to. Applicable and required only if `index_type` is explicitly `custom` or defaults to `custom`.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:483]: > Applicable and required only if `index_type` is explicitly `custom` or defaults to `custom`.

Perhaps reword to:

> Applicable and required only when the `index_type` is `custom`.

File: _clients/data-prepper/data-prepper-reference.md (hunk 4 above)
Comment (Naarcha-AWS) [inline _clients/data-prepper/data-prepper-reference.md:118]:
```suggestion
unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]
A few small tweaks.

File: _clients/data-prepper/data-prepper-reference.md [hunk 6]
```diff
@@ -19,8 +19,11 @@ keyStorePassword | No | String | Password for keystore. Optional, defaults to em
...
 metricRegistries | No | List | Metrics registries for publishing the generated metrics. Currently supports Prometheus and CloudWatch. Defaults to Prometheus.
+metricTags | No | Map | Key-value pairs as common metric tags to metric registries. The maximum number of pairs is 3. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Its value could also be set through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, the value will overwrite those set through the above methods.
+authentication | No | Object | Authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server will not perform authentication.
 processorShutdownTimeout | No | Duration | Time given to processors to clear any in-flight data and gracefully shutdown. Default is 30s.
 sinkShutdownTimeout | No | Duration | Time given to sinks to clear any in-flight data and gracefully shutdown. Default is 30s.
+peer_forwarder | No | Ojbect | Peer forwarder configurations. See [Peer forwarder options](#peer-forwarder-options) for more details.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:26]: Spelling: "Ojbect" -> "Object"

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: _clients/data-prepper/data-prepper-reference.md (hunk 1 above)
Comment (oeyh) [reply _clients/data-prepper/data-prepper-reference.md]: Thanks! All addressed in [ca1e7da](https://github.com/opensearch-project/documentation-website/pull/1469/commits/ca1e7da0795d2621ec2dd11045155b545cc8fa99).

Comment (oeyh) [reply _clients/data-prepper/data-prepper-reference.md]: Thanks! These has been addressed in [ca1e7da](https://github.com/opensearch-project/documentation-website/pull/1469/commits/ca1e7da0795d2621ec2dd11045155b545cc8fa99).

**Review by dlvenable — APPROVED** [review]
Thanks!

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by dlvenable** [comment]
@Naarcha-AWS , Can we merge this in now? We'd like to create a PR which merges our working branch `data-prepper-2.0` into `main` prior to our release today.

File: _clients/data-prepper/data-prepper-reference.md [hunk 7]
```diff
@@ -109,6 +114,8 @@ This is a source plugin that supports HTTP protocol. Currently ONLY support Json
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 port | No | Integer | The port the source is running on. Default is `2021`. Valid options are between `0` and `65535`.
+health_check_service | No | Boolean | Enables health check service on `/health` endpoint on the defined port. Default is `false`.
+unauthenticated_health_check | No | Boolean | Determines whether authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.
```
Comment (alicejw-aws) [inline _clients/data-prepper/data-prepper-reference.md:118]:
```suggestion
unauthenticated_health_check | No | Boolean | Determines whether or not authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.
```

File: _clients/data-prepper/data-prepper-reference.md [hunk 8]
```diff
@@ -472,12 +480,10 @@ socket_timeout | No | Integer | the timeout in milliseconds for waiting for data
...
-trace_analytics_raw | No | Boolean | Deprecated in favor of `index_type`. Whether to export as trace data to the `otel-v1-apm-span-*` index pattern (alias `otel-v1-apm-span`) for use with the Trace Analytics OpenSearch Dashboards plugin. Default is `false`.
-trace_analytics_service_map | No | Boolean | Deprecated in favor of `index_type`. Whether to export as trace data to the `otel-v1-apm-service-map` index for use with the service map component of the Trace Analytics OpenSearch Dashboards plugin. Default is `false`.
-index | No | String | Name of the index to export to. Only required if you don't use the `trace-analytics-raw` or `trace-analytics-service-map` presets. In other words, this parameter is applicable and required only if index_type is explicitly `custom` or defaults to `custom`.
+index | Conditionally | String | Name of the index to export to. Applicable and required only when the `index_type` is `custom`.
```
Comment (alicejw-aws) [inline _clients/data-prepper/data-prepper-reference.md:483]:
```suggestion
index | Conditionally | String | Specifies the export index name. Applicable and required only when the `index_type` is `custom`.
```

Comment (alicejw-aws) [inline _clients/data-prepper/data-prepper-reference.md:483]:
```suggestion
index | Conditionally | String | Name of the export index. Applicable and required only when the `index_type` is `custom`.
```

**Review by alicejw-aws — APPROVED** [review]
LGTM

**Pull request merged**